src/Symfony/Component/Lock/Tests/Store/MysqlStoreTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Symfony\Component\Lock\Tests\Store;
+
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\PersistingStoreInterface;
+use Symfony\Component\Lock\Store\MysqlStore;
+
+class MysqlStoreTest extends AbstractStoreTest
+{
+    protected function getEnv(): array
+    {
+        if (!$host = getenv('MYSQL_HOST')) {
+            $this->markTestSkipped('Missing MYSQL_HOST env variable');
+        }
+
+        if (!$user = getenv('MYSQL_USERNAME')) {
+            $this->markTestSkipped('Missing MYSQL_USERNAME env variable');
+        }
+
+        if (!$pass = getenv('MYSQL_PASSWORD')) {
+            $this->markTestSkipped('Missing MYSQL_PASSWORD env variable');
+        }
+
+        return [$host, $user, $pass];
+    }
+
+    protected function getPdo(): \PDO
+    {
+        [$host, $user, $pass] = $this->getEnv();
+
+        return new \PDO('mysql:host='.$host, $user, $pass);
+    }
+
+    protected function getStore(): PersistingStoreInterface
+    {
+        return new MysqlStore($this->getPdo());
+    }
+
+    public function testDriverRequirement()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        new MysqlStore(new \PDO('sqlite::memory:'));
+    }
+
+    public function testExceptionModeRequirement()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $pdo = $this->getPdo();
+        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_SILENT);
+        new MysqlStore($pdo);
+    }
+
+    public function testOtherConflictException()
+    {
+        $storeA = $this->getStore();
+        $storeA->save(new Key('foo'));
+
+        $storeB = $this->getStore();
+
+        try {
+            $storeB->save(new Key('foo'));
```

Comment:
The test is incorrect. You are testing both:
- 2 instances of Key
- 2 instances of store

You don't know why the `storeB->save` throws an exception.

For the record the below code is OK:

```
$resource = uniqid(__METHOD__, true);
$key1 = new Key($resource);
$key2 = new Key($resource);

$storeA->save($key1);
$storeb->save($key1);
```

Because stores are stateless, and you are allowed to use 2 instances of store to manipulate the same lock.